### PR TITLE
Support @KafkaPollAndInvoke annotation to replace @Trace when just using the Kafka scenario agent plugin

### LINKED ISSUE #5302: Support @KafkaOnMessage annotation to replace @Trace when just using the Kafka scenario
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [x] Feature or performance improvement

___
### Requirement or improvement
- Please describe your requirements or improvement suggestions.
For [FAQ:kafka-plugin](https://github.com/apache/skywalking/blob/master/docs/en/FAQ/kafka-plugin.md)
If we were just using the Kafka framework, then we would have to manually introduce the @trace annotation, while there are still have two problems, there would be a lot of localspan segments because poll() is empty and the link is not nice. I want to add the @KafkaOnMessage annotation to replace @Trace in this scenario to fix both of these problems

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

Fixed #5302

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java [hunk 1]
```diff
@@ -66,9 +66,10 @@ public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allA
         //
         if (records.size() > 0) {
             ConsumerEnhanceRequiredInfo requiredInfo = (ConsumerEnhanceRequiredInfo) objInst.getSkyWalkingDynamicField();
-            if (ContextManager.getRuntimeContext().get(Constants.SPRING_KAFKA_FLAG) != null) {
-                ContextManager.createEntrySpan(Constants.SPRING_KAFKA_POLL_AND_INVOKE_OPERATION_NAME, null);
-                ((SpringKafkaContext) ContextManager.getRuntimeContext().get(Constants.SPRING_KAFKA_FLAG)).setNeedStop(true);
+            KafkaContext context = (KafkaContext) ContextManager.getRuntimeContext().get(Constants.KAFKA_FLAG);
+            if (context != null) {
+                ContextManager.createEntrySpan(context.getOperationName(), null);
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java:71]: No Context Carrier?

File: apm-sniffer/apm-sdk-plugin/kafka-commons/src/main/java/org/apache/skywalking/apm/plugin/kafka/define/KafkaContext.java [hunk 2]
```diff
@@ -19,19 +19,30 @@
...
-public class SpringKafkaContext {
+public class KafkaContext {
 
-    public SpringKafkaContext() {
+    public KafkaContext(String operationName) {
         needStop = false;
+        this.operationName = operationName;
     }
 
     private boolean needStop;
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/kafka-commons/src/main/java/org/apache/skywalking/apm/plugin/kafka/define/KafkaContext.java:29]: I rechecked the codes, this seems not necessary. You always `needStop=false`. Could we consider to remove it?

File: test/plugin/scenarios/kafka-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/kafka/controller/CaseController.java [hunk 3]
```diff
@@ -223,29 +235,33 @@ public void run() {
...
-                }
+            while (true) {
+                if (pollAndInvoke(consumer)) break;
+            }
+            consumer.close();
+        }
 
-                ConsumerRecords<String, String> records = consumer.poll(100);
+        @KafkaPollAndInvoke
+        private boolean pollAndInvoke(KafkaConsumer<String, String> consumer) {
```
Comment (wu-sheng) [inline test/plugin/scenarios/kafka-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/kafka/controller/CaseController.java:245]: Are you forcing to create 2 consumer methods?

Comment (zhaoyuguang) [reply test/plugin/scenarios/kafka-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/kafka/controller/CaseController.java:245]: yes

File: apm-sniffer/apm-sdk-plugin/kafka-commons/src/main/java/org/apache/skywalking/apm/plugin/kafka/define/KafkaContext.java (hunk 2 above)
Comment (zhaoyuguang) [reply apm-sniffer/apm-sdk-plugin/kafka-commons/src/main/java/org/apache/skywalking/apm/plugin/kafka/define/KafkaContext.java]: fixed

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java (hunk 1 above)
Comment (zhaoyuguang) [reply apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java:71]: yes

File: apm-application-toolkit/apm-toolkit-kafka/pom.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    <groupId>org.apache.skywalking</groupId>
+    <version>8.2.0-SNAPSHOT</version>
+</parent>
+<modelVersion>4.0.0</modelVersion>
+
+<artifactId>apm-toolkit-kafka</artifactId>
+<packaging>jar</packaging>
+
+<url>http://maven.apache.org</url>
+</project>
```
Comment (wu-sheng) [inline apm-application-toolkit/apm-toolkit-kafka/pom.xml:32]: This pom seems not formated right. :)

File: apm-application-toolkit/pom.xml [hunk 5]
```diff
@@ -31,6 +31,7 @@
     </properties>
 
     <modules>
+        <module>apm-toolkit-kafka</module>
```
Comment (wu-sheng) [inline apm-application-toolkit/pom.xml:34]: Put in the last one.

Comment (zhaoyuguang) [reply apm-application-toolkit/pom.xml]: fixed

File: apm-application-toolkit/apm-toolkit-kafka/pom.xml (hunk 4 above)
Comment (zhaoyuguang) [reply apm-application-toolkit/apm-toolkit-kafka/pom.xml:33]: fixed,  :)

**Pull request merged**